python/ray/train/lightning/lightning_trainer.py

```diff
@@ -0,0 +1,374 @@
+import os
+import ray
+from inspect import isclass
+from typing import Any, Dict, Optional, Type
+import shutil
+
+from ray.air import session
+from ray.air.constants import MODEL_KEY
+from ray.air.config import DatasetConfig, RunConfig, ScalingConfig
+from ray.air.checkpoint import Checkpoint
+from ray.data.preprocessor import Preprocessor
+from ray.train.trainer import GenDataset
+from ray.train.torch import TorchTrainer
+from ray.train.torch.config import TorchConfig
+from ray.util import PublicAPI
+from ray.train.lightning._lightning_utils import (
+    RayDDPStrategy,
+    RayEnvironment,
+    RayDataModule,
+)
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import ModelCheckpoint
+from pytorch_lightning.plugins.environments import ClusterEnvironment
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+@PublicAPI(stability="alpha")
+class LightningConfigBuilder:
```

Comment:
can we add some small tests for this builder?

Reply:
Done